index.js

```diff
@@ -0,0 +1,43 @@
+import {
+  NativeEventEmitter,
+  NativeModules,
+  Platform,
+} from 'react-native';
+
+const { InAppUtils } = NativeModules;
+
+const InAppUtilsEmitter = new NativeEventEmitter(InAppUtils);
+
+const promisify = fn => (...args) => new Promise((resolve, reject) => {
+  fn(...args, (err, res) => err ? reject(err) : resolve(res));
+});
+
+const IAU = Platform.select({
+  ios: {
+    loadProducts: (products, cb) => cb
+      ? InAppUtils.loadProducts(products, cb)
+      : promisify(InAppUtils.loadProducts)(products),
+
+    canMakePayments: cb => cb
```

Comment:
If `canMakePayments` returns true, your `promisfy` function will reject the promise. You can create a special function just for `canMakePayments` like this one:

``` javascript
const promisifyBool = fn => (...args) => new Promise((resolve, reject) => {
    fn(...args, res => res === true || false ? resolve(res) : reject(res))
})
```

Reply:
How about this:
```
const promisify = fn => (...args) => new Promise((resolve, reject) => {
  fn(...args, (err, res) => {
    if (err !== undefined && err instanceof Error) reject(err);
    // If only one argument is given and it's not an error
    if (err !== undefined && res === undefined) resolve(err);
    resolve(res);
  });
});
```

Reply:
Looks good.

Reply:
Great! Thanks for catching that!

Reply:
@superandrew213 just realized that `err` is not always of type `Error`. When calling `restorePurchases` if the user cancels `err` comes back as a string "restore_failed". Seems the rejection types are bit inconsistent.

Reply:
Hmm ... it would be best if we enforce an error type. We can use `callback(@[RCTMakeError(@"restore_failed", nil, nil)]);` where `restore_failed` would be the error message. This will be a breaking change though.

Reply:
If we have to do a breaking change, lets keep it simple and just change `canMakePayments` to `canMakePayments(err, res)`. @chirag04 what do you think? It would also be good to use error objects too.